skbio/metadata/_interval.py

```diff
@@ -0,0 +1,274 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from ._intersection import IntervalTree
+
+
+class Interval():
```

Comment:
I've added the key functions like `__contains__`  and `__hash__`.  I didn't add `__iter__` or `__len__` because it isn't clear to me what this would return?  Iterator/length over intervals, or key/val pairs?